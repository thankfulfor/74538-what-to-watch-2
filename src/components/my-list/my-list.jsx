import * as React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import Header from '../header/header.jsx';
import {FilmList} from '../film-list/film-list-1.jsx';
import {Footer} from '../footer/footer.jsx';
import {Operation} from '../../operations/operation.js';

export const MyList = (props) => {
  const {favoriteFilms, onLoadFavoriteMovies} = props;
  onLoadFavoriteMovies();

  if (favoriteFilms === undefined) {
    return null;
  }

  return (
    <React.Fragment>

      <div className="user-page">
        <Header title="My list"/>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <FilmList films={favoriteFilms} />
        </section>
        <Footer />
      </div>
    </React.Fragment>
  );
};

MyList.propTypes = {
  favoriteFilms: PropTypes.array.isRequired,
  history: PropTypes.object.isRequired,
  onLoadFavoriteMovies: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    favoriteFilms: state.favoriteFilms,
  });
};

const mapDispatchToProps = (dispatch) => ({
  onLoadFavoriteMovies: bindActionCreators(Operation.loadFavoriteFilms, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(MyList);
