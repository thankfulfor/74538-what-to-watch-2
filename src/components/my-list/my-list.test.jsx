import React from 'react';
import {MyList} from './my-list.jsx';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

it(`MyList корректно рендерится после перезапуска`, () => {
  const historyMock = {push: jest.fn()};
  const handleLoad = jest.fn();
  const tree = shallow(<MyList onLoadFavoriteMovies={handleLoad} favoriteFilms={[]} history={historyMock} />);
  expect(toJson(tree)).toMatchSnapshot();
});
