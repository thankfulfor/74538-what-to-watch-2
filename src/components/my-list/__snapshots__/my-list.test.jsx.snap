// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList корректно рендерится после перезапуска 1`] = `
<Fragment>
  <div
    className="user-page"
  >
    <Connect(Header)
      title="My list"
    />
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <FilmList
        films={Array []}
      />
    </section>
    <Footer />
  </div>
</Fragment>
`;
